flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -273,7 +276,11 @@ public TaskExecutor(
 		this.resourceManagerConnection = null;
 		this.currentRegistrationTimeoutId = null;
 
-		this.stackTraceSampleService = new StackTraceSampleService(rpcService.getScheduledExecutor());
+		final Configuration config = taskManagerConfiguration.getConfiguration();
+		this.taskBackPressureSampleService = new BackPressureSampleService(
```

Comment:
I suggest constructing the `BackPressureSampleService` out of the constructor to make it clean, also we could avoid `taskManagerConfiguration.getConfiguration()` from outside call. It is elegant to prepare everything outside and then pass them directly into the constructor.